EIP-4844: Hook for verify_kgz_proof

mariaKt opened the pull request:
This PR adds [c-kzg-4844](https://github.com/ethereum/c-kzg-4844/tree/d1168fc791b910286c7308f7dd99fa566567e942) as a submodule and implements a a hook for the function `verify_kzg_proof`. This is part of implementing the point-evaluation-precompile of [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile).

We adapt some of the changes made in [this](https://github.com/runtimeverification/blockchain-k-plugin/pull/208) PR.

dwightguth reviewed (commented):
looks good. I will get RV to review.

anvacaru commented on plugin/krypto.md:
````diff
@@ -112,5 +112,13 @@ The BN128 elliptic curve is defined over 2-dimensional points over the fields of
+
+```k
+    syntax Bool ::= verifyKZGProof ( Bytes, Bytes, Bytes, Bytes ) [function, hook(KRYPTO.verifyKZGProof)]
+ // ---------------------------------------------------------------------------------------------------
````
formatting
```suggestion
 // -----------------------------------------------------------------------------------------------------
```

anvacaru reviewed (approved):
Looks good to me! Approved with a single comment.

mariaKt commented:
I made the changes requested by @anvacaru and @ehildenb . I also added a comment regarding the inputs of the new hook, and removed some checks that were not needed. This is again ready for review.
